fatukunda/go-party
# #168071825 Activate User

#### What does this PR do?
Sends an activation link to the user to verify their account
#### Description of Task to be completed?
 - Install `Sendgrid` to be able to send emails.
 - Create an emails folder and an `account.js` file to configure a confirmation email.
 - Add the above functionality just after the user has finished registration.
 - Create an endpoint to verify the link that was sent to the user's email.
 - Write tests.
#### How should this be manually tested?
 - Clone the repo by running `git clone https://github.com/fatukunda/go-party.git` on your terminal.
 - Install dependencies by running `yarn install`
 - Start the server by running `yarn dev`
 - Create a new user by heading to `localhost:8000/api/v1/users` and sending a `POST` request. Remember to provide the `username, email, and password` in the body. Also, make sure that you provide an active email.
 - Send the request and once it's successful, check your email to ensure that you received an account verification link.
 - Click on that link to activate your account
 - Head over to `localhost:8000/api/v1/users/login` and provide the `username` and `password` that you used while registering. If all goes well, You should successfully be logged in. 
#### What are the relevant pivotal tracker stories?
[#168071825](https://www.pivotaltracker.com/story/show/168071825)
#### Any background context you want to add?
 - None
#### Screenshots
 - Register user

<img width="1132" alt="Screen Shot 2019-09-10 at 3 00 00 PM" src="https://user-images.githubusercontent.com/4363632/64612259-0a43eb80-d3dc-11e9-96c3-edeab8ca2484.png">
 - Get verified
<img width="1132" alt="Screen Shot 2019-09-10 at 3 00 43 PM" src="https://user-images.githubusercontent.com/4363632/64612301-1d56bb80-d3dc-11e9-9d5d-ff107d713e97.png">
 - Login

<img width="1132" alt="Screen Shot 2019-09-10 at 3 01 13 PM" src="https://user-images.githubusercontent.com/4363632/64612321-29427d80-d3dc-11e9-8593-361aeba7370f.png">
